src/test/java/com/hillel/items_exchange/controller/AuthControllerIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenUserEmailExists_shouldReturnUnprocessableEntityAndThrowUnprocessableEntityException()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithExistentEmail();
+        MvcResult result = this.mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isUnprocessableEntity())
+                .andReturn();
+
+        assertThat(result.getResolvedException(), is(instanceOf(UnprocessableEntityException.class)));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenUsernameExists_shouldReturnUnprocessableEntityAndThrowUnprocessableEntityException()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithExistentUsername();
+        MvcResult result = this.mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isUnprocessableEntity())
+                .andReturn();
+
+        assertThat(result.getResolvedException(), is(instanceOf(UnprocessableEntityException.class)));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenDifferentConfirmPassword_shouldReturnBadRequestAndThrowBadRequestException()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithDifferentPasswords();
+        MvcResult result = this.mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isBadRequest())
+                .andReturn();
+
+        assertThat(result.getResolvedException(), is(instanceOf(BadRequestException.class)));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenPasswordInvalid_shouldReturnReturnNotFound()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithInvalidPassword();
+        mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isNotFound());
```

Comment:
You get wrong code here use 406 instead of it.

Reply:
I do not know how to do this, because this status returns by Spring when we have an error in annotations (validators)

Reply:
You have to change ResourceNotFoundException to another one (someValidationException) into AuthController that will return 406 code. It's not a response of Spring's validators but from our controller.